src/khoj/routers/indexer.py

```diff
@@ -0,0 +1,237 @@
+from khoj.processor.plaintext.plaintext_to_jsonl import PlaintextToJsonl
+from khoj.utils.rawconfig import ContentConfig
+from khoj.search_type import text_search, image_search
+from khoj.utils.config import SearchModels
+from khoj.utils.helpers import LRU
+from khoj.utils.rawconfig import (
+    ContentConfig,
+)
+from khoj.search_filter.date_filter import DateFilter
+from khoj.search_filter.word_filter import WordFilter
+from khoj.search_filter.file_filter import FileFilter
+from khoj.utils.config import (
+    ContentIndex,
+    SearchModels,
+)
+
+logger = logging.getLogger(__name__)
+
+indexer = APIRouter()
+
+
+class IndexBatchRequest(BaseModel):
+    org: Optional[dict[str, str]]
+    pdf: Optional[dict[str, str]]
+    plaintext: Optional[dict[str, str]]
+    markdown: Optional[dict[str, str]]
+
+
+@indexer.post("/batch")
+async def index_batch(
+    request: Request,
+    x_api_key: str = Header(None),
+    regenerate: bool = False,
+    search_type: Optional[state.SearchType] = None,
+):
+    if x_api_key != "secret":
+        raise HTTPException(status_code=401, detail="Invalid API Key")
+    state.config_lock.acquire()
+    try:
+        logger.info(f"Received batch indexing request")
+        index_batch_request_acc = ""
+        async for chunk in request.stream():
+            index_batch_request_acc += chunk.decode()
+        index_batch_request = IndexBatchRequest.parse_raw(index_batch_request_acc)
+        logger.info(f"Received batch indexing request size: {len(index_batch_request.dict())}")
+
+        # Extract required fields from config
+        state.content_index = configure_content(
+            state.content_index,
+            state.config.content_type,
+            index_batch_request.dict(),
+            state.search_models,
+            regenerate=regenerate,
+            t=search_type,
+        )
+
+    except Exception as e:
+        logger.error(f"Failed to process batch indexing request: {e}")
+    finally:
+        state.config_lock.release()
+    return Response(content="OK", status_code=200)
+
+
+def configure_content(
+    content_index: Optional[ContentIndex],
+    content_config: Optional[ContentConfig],
+    files: Optional[dict[str, dict[str, str]]],
+    search_models: SearchModels,
+    regenerate: bool = False,
+    t: Optional[state.SearchType] = None,
+) -> Optional[ContentIndex]:
+    # Run Validation Checks
+    if content_config is None:
+        logger.warning("🚨 No Content configuration available.")
+        return None
+    if content_index is None:
+        content_index = ContentIndex()
+
+    if files is None:
+        logger.warning(f"🚨 No files to process for {t.value} search.")
+        return None
+
+    try:
+        # Initialize Org Notes Search
+        if (
+            (t == None or t.value == state.SearchType.Org.value)
+            and content_config.org
+            and search_models.text_search
+            and files["org"]
```

Comment:
Wouldn't this fail in the scenario where no org-mode files exist (anymore) in the `files` dictionary? If so, this would affect the configuration of other content types below as well

Reply:
Sure, let me use `.get` instead.